pyndl/ndl_parallel.pyx

```diff
@@ -27,6 +27,18 @@ test_index *= test_outcome_index
 test_index += test_cue_index
 assert test_index == 18446744069414584320
 
+cdef enum ErrorCode:
+    NO_ERROR = 0
+    MAGIC_NUMBER_DOES_NOT_MATCH = 1
+    VERSION_NUMBER_DOES_NOT_MATCH = 2
+    INITIAL_ERROR_CODE = 3
+
+ERROR_CODES = """
```

Comment:
I suggest adding a `switch` statement for the error codes to print appropriate error messages each, but this solution seems fine for me for now!